src/protocol/vm/adapter_contract.rs

```diff
@@ -105,15 +105,15 @@ where
                         .first()
                         .cloned()
                         .ok_or_else(|| {
-                            SimulationError::DecodingError("There wasn't a calculated price".into())
+                            SimulationError::FatalError("There wasn't a calculated price".into())
```

Comment:
This error message doesn't make any sense to me can we formulate this more clearly?

Reply:
I would go further than @louise-poole and suggest adding which function is raising the error in all of these adapter contract errors